src/hssm/register.py

```diff
@@ -0,0 +1,84 @@
+"""Module for registering custom models in HSSM."""
+
+from pprint import pformat
+from typing import Dict
+
+from .defaults import (
+    DefaultConfig,
+    default_model_config,
+)
+
+
+def register_model(name: str, config: DefaultConfig) -> None:
```

Comment:
I can be convinced otherwise, but I prefer to actually have the fields in the function signature, and when people have a `DefaultConfig`, they can simply pass `**config` to the function. I think it's more explicit to document the types of the argument